core/data/permission/PermissionHelper.php

```diff
@@ -38,9 +38,14 @@ public function filterByPermission(array $resourceIds, string $right): array
         if (!in_array($right, $provider->getSupportedRights())) {
             return $resourceIds;
         }
-        $permissions = $provider->getPermissions($this->getCurrentUser(),$resourceIds);
+        $permissions = $provider->getPermissions($this->getCurrentUser(), array_keys($resourceIds));
```

Comment:
This is a tricky thing, we are assuming that the resourceIds now have some special key, which was not required before. This is dangerous can invisible by client of this class.

We should add this as another parameter in the method `filterByPermission` or pass objects instead of raw array.

Another option is to create a new method here with proper implementation